pkg/cmd/server/bootstrappolicy/infra_sa_policy.go

```diff
@@ -639,4 +642,36 @@ func init() {
 		panic(err)
 	}
 
+	err = InfraSAs.addServiceAccount(
+		InfraEndpointControllerServiceAccountName,
+		authorizationapi.ClusterRole{
+			ObjectMeta: kapi.ObjectMeta{
+				Name: EndpointControllerRoleName,
+			},
+			Rules: []authorizationapi.PolicyRule{
+				// Watching services and pods
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("get", "list", "watch"),
+					Resources: sets.NewString("services", "pods"),
+				},
+				// Managing endpoints
+				{
+					APIGroups: []string{kapi.GroupName},
+					Verbs:     sets.NewString("get", "list", "create", "update", "delete"),
```

Comment:
It doesn't. If the endpoint exists but is wrong it just overwrites it with Update().